src/Microsoft.DotNet.XHarness.iOS.Shared/Hardware/ISimulatorDevice.cs

```diff
@@ -58,8 +89,8 @@ public interface ISimulatorLoader : IDeviceLoader
         IEnumerable<SimDeviceType> SupportedDeviceTypes { get; }
         IEnumerable<SimulatorDevice> AvailableDevices { get; }
         IEnumerable<SimDevicePair> AvailableDevicePairs { get; }
-        Task<ISimulatorDevice[]> FindSimulators(TestTarget target, ILog log, bool create_if_needed = true, bool min_version = false);
+        Task<(ISimulatorDevice? Simulator, ISimulatorDevice? CompanionSimulator)> FindSimulators(TestTarget target, ILog log, bool createIfNeeded = true, bool minVersion = false);
```

Comment:
Ideally, this would return non-nullable `ISimulatorDevice Simulator` and throw when it cannot find it. But that might break Xamarin as it changes the behavior.

@mandel-macaque what do you think?

Reply:
I made the change that `FindSimulators` throws when it fails to find the simulator because it makes the `SimulatorEnumerable` work well actually

Reply:
I checked the Xamarin code and throwing shouldn't be a problem because we do it anyway if simulator comes back as null